test/integration/Cover/createStakingPool.js

```diff
@@ -121,4 +122,107 @@ describe('createStakingPool', function () {
       ),
     ).to.be.revertedWith('Caller is not a member');
   });
+
+  it("should fail to create a pool with a product that doesn't exist", async function () {
+    const { cover } = this.contracts;
+    const [manager] = this.accounts.members;
+    const { DEFAULT_PRODUCTS } = this;
+
+    const nonExistingProduct = { ...DEFAULT_PRODUCTS[0], productId: 500 };
+
+    await expect(
+      cover.connect(manager).createStakingPool(
+        false, // isPrivatePool,
+        DEFAULT_POOL_FEE, // initialPoolFee
+        DEFAULT_POOL_FEE, // maxPoolFee,
+        [nonExistingProduct], // products
+        '', // ipfsDescriptionHash
+      ),
+    ).to.be.revertedWithPanic(errors.INVALID_ARRAY_ACCESS);
+    // TODO: do we want an explicit error for the non-existing product?
```

Comment:
Note that there is no explicit existence check, but it currently tries to index the products array at an invalid index and reverts.